### ISSUE TITLE
Type.createInstance floating crash

### ISSUE THREAD

**Issue by Yanrishatum** [HaxeFoundation/hashlink#203]
HL: 1.8.0
Haxe (x64): 4.0.0-preview.5+7eb789f54 and 4.0.0-preview.5+6e1e8f374
OS: Win10x64
I cannot provide the reproduction code, because I never managed to isolate the bug and reproduce it outside of source code. (I can try to package all dependencies and source together if required).
The `cval` inside `Reflect.callMethod` is always `Failed to read @0x300020:4` with address obviously changing.
This often happens when I use Heaps `hxd.Res`, since it uses reflection to load resource (`loadCache`). And honestly I don't even understand how I fix it, because sometimes it's enough to add one line of code somewhere else for it to disappear (and then reappear on loading completely different resource).
The compiled .hl file that crashes attached. 
[crashing.zip](https://github.com/HaxeFoundation/hashlink/files/2591591/crashing.zip)

Callstack:
```
Access violation
Called from $Reflect.callMethod (E:\HaxeToolkit\haxe\std\hl/_std/Reflect.hx line 83)
Called from $Type.createInstance (E:\HaxeToolkit\haxe\std\hl/_std/Type.hx line 134)
Called from hxd.res.Loader.loadCache (hxd/res/Loader.hx line 45)
Called from Main.init (Main.hx line 35)
Called from local function #1790 (hxd/App.hx line 116)
Called from engine.LDEngine.loadAssets (engine/LDEngine.hx line 72)
Called from hxd.App.setup (hxd/App.hx line 114)
Called from h3d.Engine.onCreate (h3d/Engine.hx line 240)
Called from local function #3425 (h3d/impl/DirectXDriver.hx line 257)
Called from local function #3828 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 140)
Called from local function #3829 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 74)
Called from haxe.$MainLoop.tick (E:\HaxeToolkit\haxe\std\haxe/MainLoop.hx line 166)
Called from hxd.$System.runMainLoop (hxd/System.hl.hx line 124)
Called from hxd.$System.runMainLoop (hxd/System.hl.hx line 122)
Called from local function #3828 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 140)
Called from local function #3829 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 74)
Called from haxe.$MainLoop.tick (E:\HaxeToolkit\haxe\std\haxe/MainLoop.hx line 166)
Called from haxe.$EntryPoint.processEvents (E:\HaxeToolkit\haxe\std\haxe/EntryPoint.hx line 104)
Called from haxe.$EntryPoint.run (E:\HaxeToolkit\haxe\std\haxe/EntryPoint.hx line 136)
```

**Comment by ncannasse** [comment]
I don't seem to be able to reproduce, I'm getting instead:
```
Resource file not found 'core/alucard_walk.png'
Called from hxd.fs.LocalFileSystem.get (hxd/fs/LocalFileSystem.hx line 425)
Called from hxd.res.Loader.load (hxd/res/Loader.hx line 36)
Called from hxd.res.GifImage.toFrames (hxd/res/GifImage.hx line 19)
Called from Main.init (Main.hx line 35)
Called from local function #1790 (hxd/App.hx line 116)
Called from engine.LDEngine.loadAssets (engine/LDEngine.hx line 72)
Called from hxd.App.setup (hxd/App.hx line 114)
Called from h3d.Engine.onCreate (h3d/Engine.hx line 240)
Called from local function #3435 (h3d/impl/DirectXDriver.hx line 257)
Called from local function #3848 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 140)
Called from local function #3849 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 74)
Called from haxe.$MainLoop.tick (E:\HaxeToolkit\haxe\std\haxe/MainLoop.hx line 166)
Called from hxd.$System.runMainLoop (hxd/System.hl.hx line 124)
Called from hxd.$System.runMainLoop (hxd/System.hl.hx line 122)
Called from local function #3848 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 140)
Called from local function #3849 (E:\HaxeToolkit\haxe\std\haxe/Timer.hx line 74)
Called from haxe.$MainLoop.tick (E:\HaxeToolkit\haxe\std\haxe/MainLoop.hx line 166)
Called from haxe.$EntryPoint.processEvents (E:\HaxeToolkit\haxe\std\haxe/EntryPoint.hx line 104)
Called from haxe.$EntryPoint.run (E:\HaxeToolkit\haxe\std\haxe/EntryPoint.hx line 136)
```

I'm using HL/GIT but there is very little differences vs 1.8
Haxe version is unrelated since I'm directly running the HL without compiling.
Are you sure you are not on HL 1.7 ? There was a bug related to callMethod wrt haxe preview 5 fixed in 1.8

**Comment by Yanrishatum** [comment]
Huh, actually now that I checked, for some reason I had HL 1.7.0 symlinked instead of 1.8.0. Seems like I'm an idiot, and forgot to set correct HL after downloading it, sorry for bothering. :)
